components/tools/OmeroPy/src/omero/plugins/import.py

```diff
@@ -176,55 +176,39 @@ def _configure(self, parser):
         parser.add_argument(
+            "--skip", choices=SKIP_CHOICES,
+            type=str, help="Optional steps to skip during import")
+        parser.add_argument(
             "path", nargs="*",
             help="Path to be passed to the Java process")
 
         parser.set_defaults(func=self.importer)
 
-    def importer(self, args):
-
-        if args.clientdir:
-            client_dir = path(args.clientdir)
-        else:
-            client_dir = self.ctx.dir / "lib" / "client"
-        etc_dir = self.ctx.dir / "etc"
-        xml_file = etc_dir / "logback-cli.xml"
-        logback = "-Dlogback.configurationFile=%s" % xml_file
-
-        try:
-            classpath = [file.abspath() for file in client_dir.files("*.jar")]
-        except OSError as e:
-            self.ctx.die(102, "Cannot get JAR files from '%s' (%s)"
-                         % (client_dir, e.strerror))
-        if not classpath:
-            self.ctx.die(103, "No JAR files found under '%s'" % client_dir)
-
-        xargs = [logback, "-Xmx1024M", "-cp", os.pathsep.join(classpath)]
-
-        # Here we permit passing ---file=some_output_file in order to
-        # facilitate the omero.util.import_candidates.as_dictionary
-        # call. This may not always be necessary.
-        out = args.file
-        err = args.errs
-
-        if out:
-            out = open(out, "w")
-        if err:
-            err = open(err, "w")
-
-        login_args = []
-        if args.javahelp:
-                login_args.append("-h")
+    def set_login_arguments(self, args):
+        """Set the connection arguments"""
 
-        if "-h" not in login_args and "-f" not in login_args \
-                and not args.java_f and not args.java_advanced_help:
+        # Connection is required unless help arguments or -f is passed
+        connection_required = ("-h" not in self.command_args and
+                               not args.java_f and
+                               not args.java_advanced_help)
+        if connection_required:
             client = self.ctx.conn(args)
-            srv = client.getProperty("omero.host")
-            prt = client.getProperty("omero.port")
-            login_args.extend(["-s", srv])
-            login_args.extend(["-p", prt])
-            login_args.extend(["-k", client.getSessionId()])
-
+            self.command_args.extend(["-s", client.getProperty("omero.host")])
+            self.command_args.extend(["-p", client.getProperty("omero.port")])
+            self.command_args.extend(["-k", client.getSessionId()])
+
+    def set_skip_arguments(self, args):
+        """Set the arguments to skip steps during import"""
+        if args.skip in ['all', 'checksum']:
+            self.command_args.append("--no_pixels_checksum")
+            self.command_args.append("--checksum_algorithm=File-Size-64")
+        if args.skip in ['all', 'thumbnails']:
+            self.command_args.append("--no_thumbnails")
```

Comment:
Is the `--no_thumbnails` option not exposed in the help? Or am I just missing it?